src/Symfony/Component/DependencyInjection/Context.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection;
+
+use Symfony\Component\DependencyInjection\Exception\ContextElementNotFoundException;
+
+/**
+ * @author Maxime Steinhausser <maxime.steinhausser@gmail.com>
+ */
+final class Context
+{
+    private $elements;
+
+    private function __construct(array $elements = array())
+    {
+        $this->elements = $elements;
+    }
+
+    /**
+     * Creates a new context allowing to share elements during the container building phase.
+     *
+     * Context elements is an array where the values are the elements to share
+     * and the keys are strings allowing to retrieve an element.
+     *
+     * For instance:
+     *
+     * Context::create(array('kernel' => new BootingKernel($kernel)));
+     *
+     * You can optionally pass an existing Context instance. Thus, original elements are reused,
+     * but new context elements replace any existing key.
+     *
+     * @param array        $contextElements An array of elements indexed by a string.
+     * @param Context|null $fromContext     An optional Context instance used as base.
+     *
+     * @return Context
+     */
+    public static function create(array $contextElements = array(), Context $fromContext = null)
```

Comment:
What about `array $elements = array(), Context $previous = null`

It doesnt repeat terms, that are obvious due the _context_ of this class.